source/material_model/utilities.cc

```diff
@@ -472,28 +472,78 @@ namespace aspect
           // Read data from disk and distribute among processes
           std::istringstream in(Utilities::read_and_distribute_file_content(filename, comm));
 
-          std::getline(in, temp); // eat first line
-          std::getline(in, temp); // eat next line
-          std::getline(in, temp); // eat next line
-          std::getline(in, temp); // eat next line
-
-          in >> min_temp;
-          std::getline(in, temp);
-          in >> delta_temp;
-          std::getline(in, temp);
-          in >> n_temperature;
-          std::getline(in, temp);
-          std::getline(in, temp);
-          in >> min_press;
-          min_press *= 1e5;  // conversion from [bar] to [Pa]
-          std::getline(in, temp);
-          in >> delta_press;
-          delta_press *= 1e5; // conversion from [bar] to [Pa]
-          std::getline(in, temp);
-          in >> n_pressure;
-          std::getline(in, temp);
-          std::getline(in, temp);
-          std::getline(in, temp);
+          // The following lines read in a PerpleX tab file in standard format
+          // First line is the Perplex version number
+          std::getline(in, temp); // get next line, table file name
+          std::getline(in, temp); // get next line, dimension of table
+          std::getline(in, temp); // get next line, either T(K) or P(bar)
+
+          for (unsigned int i=0; i<2; i++)
+            {
+              std::string natural_variable;
+              in >> natural_variable;
+
+              if (natural_variable[0] == 'T')
+                {
+                  std::getline(in, temp);
+                  in >> min_temp;
+                  std::getline(in, temp);
+                  in >> delta_temp;
+                  std::getline(in, temp);
+                  in >> n_temperature;
+                  std::getline(in, temp); // get next line, either T(K), P(bar) or number of columns
+                }
+              else if (natural_variable[0] == 'P')
+                {
+                  std::getline(in, temp);
+                  in >> min_press;
+                  min_press *= 1e5;  // conversion from [bar] to [Pa]
+                  std::getline(in, temp);
+                  in >> delta_press;
+                  delta_press *= 1e5; // conversion from [bar] to [Pa]
```

Comment:
Isn't that factor 1.014e5?

Reply:
Turns out that wikipedia says that 1 bar = 10^5 Pa indeed, and so slight less than one standard atmosphere (which is 1.014 bar).

Reply:
Some of the old low pressure papers report their pressures in mmHg... it's a fun world!!